wakuv2/waku.go

```diff
@@ -390,10 +392,8 @@ func (w *Waku) getDiscV5BootstrapNodes(ctx context.Context, addresses []string)
 			wg.Add(1)
 			go func(addr string) {
 				defer wg.Done()
-				if err := w.dnsDiscover(ctx, addr, retrieveENR); err != nil {
-					mu.Lock()
-					w.seededBootnodesForDiscV5 = false
-					mu.Unlock()
+				if err := w.dnsDiscover(ctx, addr, retrieveENR, useOnlyDnsDiscCache); err != nil {
+					go w.retryDnsDiscoveryWithBackoff(ctx, addr, w.dnsDiscAsyncRetrievedSignal)
```

Comment:
Looks like a `wg.Add(1)` is needed here. Otherwise might panic on stopping the node, e.g. accessing `w.ctx`.